File: src/main/java/com/jozufozu/exnihiloomnia/common/world/WorldTypeSkyblock.java
package com.jozufozu.exnihiloomnia.common.world;

import com.jozufozu.exnihiloomnia.common.ModConfig;
import net.minecraft.world.World;
import net.minecraft.world.WorldType;
import net.minecraft.world.gen.IChunkGenerator;

public class WorldTypeSkyblock extends WorldType
{
    public static WorldType SKY_BLOCK;
    
    public WorldTypeSkyblock()
    {
        super("exnihiloomnia");
    }
    
    @Override
    public float getCloudHeight()
    {
        return ModConfig.world.cloudLevel;
    }
    
    @Override
    public int getMinimumSpawnHeight(World world)
    {
        return ModConfig.world.spawnLevel;
    }
    
    @Override
    public IChunkGenerator getChunkGenerator(World world, String generatorOptions)
    {
        return new ChunkGeneratorSkyBlock(world, world.getSeed());
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/world/WorldTypeSkyblock.kt
package com.jozufozu.exnihiloomnia.common.world

import com.jozufozu.exnihiloomnia.common.ModConfig
import net.minecraft.world.World
import net.minecraft.world.WorldType
import net.minecraft.world.gen.IChunkGenerator

class WorldTypeSkyblock : WorldType("exnihiloomnia") {

    override fun getCloudHeight(): Float {
        return ModConfig.world.cloudLevel.toFloat()
    }

    override fun getMinimumSpawnHeight(world: World): Int {
        return ModConfig.world.spawnLevel
    }

    override fun getChunkGenerator(world: World, generatorOptions: String): IChunkGenerator {
        return ChunkGeneratorSkyBlock(world, world.seed)
    }

    companion object {
        var SKY_BLOCK: WorldType? = null
    }
}
